File: app/frontend/src/types/Resource.ts
export type Resource = 'water_tank' | 'borehole' | 'trailer' | 'dam' | 'aircraft' | 'crew' | 'other';

export type Capacity = 'liters' | 'members' | 'other';

export type Status = 'available' | 'dispatched' | 'unavailable';

export interface ResourceTable {
    id: string;
    resource: Resource;
    otherResource: string;
    capacity: number;
    capacityUnit: Capacity;
    otherCapacity: string;
    status: Status;
    availableFrom: string;
    availableUntil: string | null;
    location: string;
    externalPin: { lat: number; lng: number };
    name: string;
    contact: string;
}

export interface ResourceList {
    data: ResourceTable[];
    total: number;
}

export type ResourceInput = Omit<ResourceTable, 'id' | 'status' | 'capacityUnit'>

File: app/backend/src/enums/resource.py
from enum import Enum

class ResourceType(str, Enum):
    water_tank = "water_tank"
    borehole = "borehole"
    trailer = "trailer"
    dam = "dam"
    aircraft = "aircraft"
    crew = "crew"
    other = "other"
    
class CapacityUnit(str, Enum):
    liters = "liters"
    members = "members"
    other = "other"
    
class ResourceStatus(str, Enum):
    available = "available"
    dispatched = "dispatched"
    unavailable = "unavailable"
    
def capacity_unit_for(resource_type: ResourceType) -> CapacityUnit:
    """
    This unit is fully determined by the resource type, so server derives it instead of trusting the client
    """
    if resource_type == ResourceType.other:
        return CapacityUnit.other
    if resource_type == ResourceType.crew:
        return CapacityUnit.members
    return CapacityUnit.liters